nni/contrib/distillation/uid_dataset.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+from collections import defaultdict
+from typing import Any, Callable, Dict, List, Type
+
+import torch
+from torch.utils.data import Dataset, IterableDataset
+
+from .utils import _default_get_rngs_state, _default_set_rngs_state, _default_manual_seed
+
+
+class _UidDataset(Dataset):
+    def __init__(self, dataset: Dataset, *args, **kwargs):
+        assert not isinstance(dataset, IterableDataset), 'Do not support `IterableDataset`.'
+        self._dataset = dataset
+        self._replay_mode = False
+
+    def __len__(self):
+        return len(self._dataset)  # type: ignore
+
+    def __getitem__(self, index):
+        raise NotImplementedError()
+
+    def get_origin_dataset(self):
+        return self._dataset
+
+    def observe(self):
+        """
+        Observe mode means this dataset is on processing by `Preprocessor`
```

Comment:
when will observe() be used?

Reply:
If user need more distillation labels by run `preprocess_labels` second time, he needs to use `observe()` to set `Preprocessor` to observe mode.